app/steps/step.rb

```diff
@@ -8,7 +8,12 @@ class Step
   class_attribute :attribute_names
 
   def self.step_attributes(*attribute_names)
-    self.attribute_names = attribute_names.map(&:to_s)
-    attr_accessor(*attribute_names)
+    self.attribute_names = attribute_names
+
+    attributes_or_keys = attribute_names.map do |attr|
+      attr.class == Symbol ? attr : attr.keys
+    end
+
+    attr_accessor(*attributes_or_keys.flatten.map(&:to_s))
   end
```

Comment:
The changes in here are to allow for not only what we've had up until now:

`"key" => "value"`

but also an array of values, which is what we're receiving when we submit N documents:

`"key" => ["value1", "value2"]`